### ISSUE TITLE
Circus tent renders wrong

### ISSUE THREAD

**Issue by xrstf** [LinusU/node-rct-graphics#1]
While testing my own code, I stumbled upon a difference in how you and me export some colors. I'm using the csg files from the RCT1 Deluxe Edition and the RCT Inspector shows a blue entrance to the circus tent, compared to a brown one in the game, my code and in another 3rd party tool.

![circus](https://cloud.githubusercontent.com/assets/127499/10268860/cd7b0f5e-6ac5-11e5-8617-c4e0eafbde03.png)

I'm not sure if this is a bug in your code or what's going on. I was using the same two remap palettes in all 4 examples.

**Comment by LinusU** [comment]
Yeah I've been banging my head against the wall with this for hours :laughing: 

I've had this picture sitting on my desktop to remind me to fix it but I haven't got around to it.

![screen shot 2015-08-16 at 02 23 27](https://cloud.githubusercontent.com/assets/189580/10269385/8c448aa4-6ad6-11e5-8f33-dc8b469b43a7.png)

I would love to know where you got your color from and how you are remapping them!

**Comment by xrstf** [comment]
I think I found your problem. Based on your function naming ("Direct image"), I guess you also took http://tid.rctspace.com/csg/csg.html as the basis for your code.

The docs mention that the colors are **[Blue, Green, Red]**. You seem to assume they are RGB, leading to wrong palettes. Changing the `loadPaletteEntries` to do

```
result.set(record.offsetX + i, [rgb[2], rgb[1], rgb[0]]);
```

fixes the colors. The odd order comes from the fact, I guess, that the files use Little Endian encoding, so it kind of makes sense to reverse the colors from RGB to BGR as well.

**Comment by LinusU** [comment]
Mind. Blown.

That is such an easy fix, I can't believe I missed it, haha. I'm gonna have a look at it this weekend, thanks for the help man!
